packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -620,8 +641,9 @@ function createArrayNodeProxy(
 				return Reflect.set(dispatch, key, newValue, proxy);
 			}
 
-			const field = getSequenceField(proxy);
-			const maybeIndex = asIndex(key, field.length);
+			// Array nodes treat all non-negative integer indexes as array access.
+			// Using Infinity here (rather than length) ensures that indexing past the end doesn't create additional session local properties.
+			const maybeIndex = asIndex(key, Number.POSITIVE_INFINITY);
```

Comment:
"get" could use a similar rework for the same reason (if for some reason someone defined an inherited member "5", it should probably get shadowed regardless of length, even on read.)

THis makes me suspect "asIndex" might not need a length at all, and separating that check out might be better.

Reply:
Agreed - I made work item 7753 to track that.